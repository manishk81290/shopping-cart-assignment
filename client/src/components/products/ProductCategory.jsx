import React from "react";
import { Link } from "react-router-dom";
import { Col, Card, ListGroup } from "react-bootstrap";

const ProductCategory = () => {
  const categoryList = [
    {
      name: "Beverages",
      key: "beverages",
      description:
        "Our beverage department will ensure your fridge is always fully stocked. Shop for soda, juice, beer and more. ",
      enabled: true,
      order: 3,
      imageUrl: "/static/images/category/beverages.png",
      id: "5b675e5e5936635728f9fc30",
    },
    {
      name: "Bakery Cakes and Dairy",
      key: "bakery-cakes-dairy",
      description:
        "The best cupcakes, cookies, cakes, pies, cheesecakes, fresh bread, biscotti, muffins, bagels, fresh coffee, milk and more.",
      enabled: true,
      order: 2,
      imageUrl: "/static/images/category/bakery.png",
      id: "5b6899123d1a866534f516de",
    },
    {
      name: "Beauty and Hygiene",
      key: "beauty-hygiene",
      description:
        "Buy beauty and personal care products online in India at best prices.",
      enabled: true,
      order: 4,
      imageUrl: "/static/images/category/beauty.png",
      id: "5b68994e3d1a866534f516df",
    },
    {
      name: "Baby Care",
      key: "baby",
      description:
        "Shop online for Baby Products, Diapers, Skin Care Products,etc.",
      enabled: true,
      order: 5,
      imageUrl: "/static/images/category/baby.png",
      id: "5b6899683d1a866534f516e0",
    },
    {
      name: "Seafood",
      key: "seafood",
      description: "Great place to buy fresh seafood.",
      enabled: false,
      order: -1,
      id: "5b68997d3d1a866534f516e1",
    },
    {
      name: "Fruits & Vegetables",
      key: "fruit-and-veg",
      description: "A variety of fresh fruits and vegetables.",
      enabled: true,
      order: 1,
      imageUrl: "/static/images/category/fruits.png",
      id: "5b6899953d1a866534f516e2",
    },
  ].map((cat) => {
    return { name: cat.name, key: cat.key, id: cat.id };
  });
  console.log(categoryList);
  return (
    <Col md={3}>
      <Card style={{ width: "100%" }}>
        {/* <Card.Header>Featured</Card.Header> */}
        <ListGroup variant="flush">
          {categoryList.map((catList) => {
            return (
              <ListGroup.Item key={catList.key}>
                <Link to={`/products/${catList.id}`}>{catList.name}</Link>
              </ListGroup.Item>
            );
          })}
        </ListGroup>
      </Card>
    </Col>
  );
};

export default ProductCategory;
