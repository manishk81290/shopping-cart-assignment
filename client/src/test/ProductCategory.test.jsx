import React from "react";
import ProductCategory from "./../components/products/ProductCategory";
import ShallowRenderer from "react-test-renderer/shallow";

let realUseContext;
let useContextMock;
const sampleContextData = {
  categories: [
    {
      name: "Beauty and Hygiene",
      key: "beauty-hygiene",
      description:
        "Buy beauty and personal care products online in India at best prices.",
      enabled: true,
      order: 4,
      imageUrl: "/static/images/category/beauty.png",
      id: "5b68994e3d1a866534f516df",
    },
  ],
  getCategories: () => {},
};

beforeEach(() => {
  realUseContext = React.useContext;
  useContextMock = React.useContext = jest.fn();
});
// Cleanup mock
afterEach(() => {
  React.useContext = realUseContext;
});

it("ProductCategory with mock useContext hook", () => {
  useContextMock.mockReturnValue(sampleContextData);
  const element = new ShallowRenderer().render(<ProductCategory />);
  expect(element).toBeTruthy();
  expect(element).toMatchSnapshot();
});
