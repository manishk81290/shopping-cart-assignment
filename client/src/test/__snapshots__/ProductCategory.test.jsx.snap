// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCategory with mock useContext hook 1`] = `
<Col
  className="bg-col-e8e8e8 pr-0 pl-0"
  md={3}
>
  <Card
    body={false}
    className="xs-hide"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <ListGroup
      variant="flush"
    >
      <React.Fragment>
        <ListGroupItem
          active={false}
          disabled={false}
        >
          <Link
            to="/products/5b68994e3d1a866534f516df"
          >
            Beauty and Hygiene
          </Link>
        </ListGroupItem>
      </React.Fragment>
    </ListGroup>
  </Card>
  <DropdownButton
    className="xs-only"
    id="dropdown-basic-button"
    title="Category Name"
  >
    <React.Fragment>
      <DropdownItem
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        disabled={false}
      >
        <Link
          to="/products/5b68994e3d1a866534f516df"
        >
          Beauty and Hygiene
        </Link>
      </DropdownItem>
    </React.Fragment>
  </DropdownButton>
</Col>
`;
